src/core/trackingmodel.h

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+ trackingmodel.h - TrackingModel
+
+ ---------------------
+ begin                : 20.02.2020
+ copyright            : (C) 2020 by David Signer
+ email                : david (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef TRACKINGMODEL_H
+#define TRACKINGMODEL_H
+
+#include <QAbstractItemModel>
+
+#include "rubberbandmodel.h"
+#include "tracker.h"
+
+class RubberbandModel;
+class Track;
+
+class TrackingModel : public QAbstractItemModel
+{
+    Q_OBJECT
+
+  public:
+    explicit TrackingModel( QObject *parent = nullptr );
+    ~TrackingModel() override;
+
+    enum TrackingRoles
+    {
+      DisplayString = Qt::UserRole,
+      VectorLayer,
+      RubberModel,
+      TimeInterval,
+      MinimumDistance,
+      Conjunction,
+      Visible,
+      Feature
+    };
+
+    QHash<int, QByteArray> roleNames() const override;
+    QModelIndex index( int row, int column, const QModelIndex &parent = QModelIndex() ) const override;
+    QModelIndex parent( const QModelIndex &index ) const override;
+    int rowCount( const QModelIndex &parent = QModelIndex() ) const override;
+    int columnCount( const QModelIndex &parent = QModelIndex() ) const override;
+    QVariant data( const QModelIndex &index, int role = Qt::DisplayRole ) const override;
+    virtual bool setData( const QModelIndex &index, const QVariant &value, int role ) override;
+
+    Q_INVOKABLE void startTracker( QgsVectorLayer *layer );
+    Q_INVOKABLE void createTracker( QgsVectorLayer *layer, bool visible );
```

Comment:
Can you document createTracker and startTracker? It is not intuitively clear to me what each of these do.